extensions/ql-vscode/test/factories/db-item-factories.ts

```diff
@@ -1,34 +1,167 @@
+  LocalListDbItem,
+  RemoteOwnerDbItem,
+  RemoteRepoDbItem,
+  RemoteSystemDefinedListDbItem,
   RemoteUserDefinedListDbItem,
+  RootLocalDbItem,
   RootRemoteDbItem,
 } from "../../src/databases/db-item";
 
-export function createRootRemoteDbItem(): RootRemoteDbItem {
+// Root Remote Db Items
+export function createRootRemoteDbItem({
+  children = [],
+  expanded = false,
+}: {
+  children?: Array<
+    | RemoteOwnerDbItem
+    | RemoteSystemDefinedListDbItem
+    | RemoteUserDefinedListDbItem
+  >;
+  expanded?: boolean;
+}): RootRemoteDbItem {
   return {
     kind: DbItemKind.RootRemote,
-    children: [],
-    expanded: false,
+    children,
+    expanded,
+  };
+}
+
+export function createRemoteOwnerDbItem({
+  ownerName = `owner${getRandomInt()}`,
+  selected = false,
+}: {
+  ownerName?: string;
+  selected?: boolean;
+}): RemoteOwnerDbItem {
+  return {
+    kind: DbItemKind.RemoteOwner,
+    selected,
+    ownerName,
+  };
+}
+
+export function createRemoteRepoDbItem({
+  repoFullName = `repoFullName${getRandomInt()}`,
+  selected = false,
+  parentListName = undefined,
+}: {
+  repoFullName?: string;
+  selected?: boolean;
+  parentListName?: string;
+}): RemoteRepoDbItem {
+  return {
+    kind: DbItemKind.RemoteRepo,
+    selected,
+    repoFullName,
+    parentListName,
+  };
+}
+
+export function createRemoteSystemDefinedListDbItem({
+  listName = `top_${getRandomInt()}`,
+  listDisplayName = `Display Name`,
+  listDescription = `Description`,
+  selected = false,
+}: {
+  listName?: string;
+  listDisplayName?: string;
+  listDescription?: string;
+  selected?: boolean;
+}): RemoteSystemDefinedListDbItem {
+  return {
+    kind: DbItemKind.RemoteSystemDefinedList,
+    selected,
+    listName,
+    listDisplayName,
+    listDescription,
   };
 }
 
 export function createRemoteUserDefinedListDbItem({
-  name = "list1",
+  expanded = false,
+  selected = false,
+  listName = `list${getRandomInt()}`,
+  repos = [
+    createRemoteRepoDbItem({
+      parentListName: listName,
+    }),
+  ],
 }: {
-  name: string;
+  listName?: string;
+  expanded?: boolean;
+  selected?: boolean;
+  repos?: RemoteRepoDbItem[];
 }): RemoteUserDefinedListDbItem {
   return {
     kind: DbItemKind.RemoteUserDefinedList,
-    selected: false,
-    expanded: false,
-    listName: name,
-    repos: [
-      {
-        kind: DbItemKind.RemoteRepo,
-        selected: false,
-        repoFullName: "repo1",
-        parentListName: name,
-      },
-    ],
+    expanded,
+    selected,
+    listName,
+    repos,
   };
 }
+
+// Root Local Db Items
+export function createRootLocalDbItem({
+  children = [],
+  expanded = false,
+}: {
+  children?: Array<LocalDatabaseDbItem | LocalListDbItem>;
+  expanded?: boolean;
+}): RootLocalDbItem {
+  return {
+    kind: DbItemKind.RootLocal,
+    children,
+    expanded,
+  };
+}
+
+export function createLocalDatabaseDbItem({
+  databaseName = `database${getRandomInt()}`,
+  dateAdded = getRandomInt(),
```

Comment:
(Optional) Since this is a number representing a date it would be great if the number was not totally random.

Reply:
Good point